payment-circle/src/main/java/org/stellar/anchor/paymentservice/circle/CirclePaymentsService.java

```diff
@@ -352,11 +390,29 @@ private void validateSendPaymentInput(@NonNull Account sourceAccount, @NonNull A
             case STELLAR:
                 sendPaymentMono = sendTransfer(sourceAccount, destinationAccount, circleBalance);
                 break;
+            case BANK_WIRE:
+                sendPaymentMono = sendPayout(sourceAccount, destinationAccount, circleBalance);
+                break;
             default:
                 throw new RuntimeException("unsupported destination network '" + destinationAccount.network + "'");
         }
 
-        return sendPaymentMono;
+        return Mono.zip(getDistributionAccountAddress(), sendPaymentMono).flatMap(args -> {
+            String distributionAccountAddress = args.getT1();
+            Payment payment = args.getT2();
+
+            // fill source account level
+            Account sourceAcc = payment.getSourceAccount();
+            Boolean isSourceEqualsDistribution = sourceAcc.id.equals(distributionAccountAddress);
+            sourceAcc.capabilities.set(Network.BANK_WIRE, isSourceEqualsDistribution);
+
+            // fill destination account level
+            Account destinationAcc = payment.getDestinationAccount();
+            Boolean isDestinationEqualsDistributionOrBankWire = destinationAcc.network.equals(Network.BANK_WIRE) || destinationAcc.id.equals(distributionAccountAddress);
+            destinationAcc.capabilities.set(Network.BANK_WIRE, isDestinationEqualsDistributionOrBankWire);
```

Comment:
💡  This chunk of code seems like it should be in its own function and called within the `sendTransfer()` and `sendPayout()` functions. Won't these functions also need to have their account capabilities set before returning?

Reply:
That makes sense to me! Updating in https://github.com/stellar/java-stellar-anchor-sdk/pull/21/commits/84130785915b811531abe15e3e11decd4575cfe5